File: Data Structures & Algorithms/anagram-groups/submission-0.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        var result = [[String]]()
        var valDic = [String: [String]]()

        for i in 0..<strs.count {
            let sorted = strs[i].sorted()
            let key = String(sorted)
            
            if var val = valDic[key] {
                valDic[key] = val + [strs[i]]
            } else {
                valDic[key] = [strs[i]]
            }
            

        }
        
        for (_, val) in valDic {
            result.append(val)
        }

        return result
    }
}

File: Data Structures & Algorithms/largest-unique-number/submission-0.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        count = {}
        for i in range(len(nums)):
            count[nums[i]] = count.get(nums[i], 0) + 1

        sorted_map = OrderedDict(sorted(count.items(), reverse=True))    


        for num, feq in sorted_map.items():
            if feq == 1:
                return num
        return -1